debug.go

```diff
@@ -0,0 +1,171 @@
+			msg += "\n"
+		}
+
+		logger.Printf(msg, a...)
+	}
+}
+
+// Prints to the standard logger if level is warn or greater; arguments are
+// handled in the manner of log.Printf, but a newline is appended.
+func warn(msg string, a ...interface{}) {
+	print(LogWarn, msg, a...)
+}
+
+// Helper function to simply warn about an error received.
+func warne(err error) {
+	warn(err.Error())
+}
+
+// Prints to the standard logger if level is status or greater; arguments are
+// handled in the manner of log.Printf, but a newline is appended.
+func status(msg string, a ...interface{}) {
+	print(LogStatus, msg, a...)
+}
+
+// Caution messages only log if the number of the same caution messages is
+// greater than the CautionThreshold, reducing the number of log messages
+// in the system but still reporting valuable information.
+//
+// NOTE: take care with string formatting individual messages, this could
+// lead to a very full caution counter that is taking up memory.
+func caution(msg string, a ...interface{}) {
+	if logLevel > LogCaution {
+		// Don't waste memory if the log level is set below caution.
+		return
+	}
+
+	cautionCounter.Lock()
+	defer cautionCounter.Unlock()
+
+	msg = fmt.Sprintf(msg, a...)
+	cautionCounter.counts[msg]++
+
+	if cautionCounter.counts[msg] >= CautionThreshold {
+		print(LogCaution, msg)
+		delete(cautionCounter.counts, msg)
+	}
+}
+
+// Prints to the standard logger if level is info or greater; arguments are
+// handled in the manner of log.Printf, but a newline is appended.
+func info(msg string, a ...interface{}) {
+	print(LogInfo, msg, a...)
+}
+
+// Prints to the standard logger if level is debug or greater; arguments are
+// handled in the manner of log.Printf, but a newline is appended.
+func debug(msg string, a ...interface{}) {
+	print(LogDebug, msg, a...)
+}
+
+// Prints to the standard logger if level is trace or greater; arguments are
+// handled in the manner of log.Printf, but a newline is appended.
+func trace(msg string, a ...interface{}) {
+	print(LogTrace, msg, a...)
+}
+
+//===========================================================================
+// Database Logger
+//===========================================================================
+
+// DBLogger implements badger.Logger to ensure database log messages get appropriately
+// logged at the level specified by Rossby.
+//
+// TODO: Generalize this structure and the rest of this package into a kansaslabs log package
+type DBLogger struct{}
+
+// Errorf writes a log message to warn
+func (l *DBLogger) Errorf(msg string, a ...interface{}) {
+	warn(msg, a...)
+}
+
+// Warningf writes a log message to caution
+func (l *DBLogger) Warningf(msg string, a ...interface{}) {
+	caution(msg, a...)
+}
+
+// Infof writes a message to info
+func (l *DBLogger) Infof(msg string, a ...interface{}) {
+	info(msg, a...)
+}
+
+// Debugf writes a message to trace
+func (l *DBLogger) Debugf(msg string, a ...interface{}) {
+	trace(msg, a...)
+}
```

Comment:
This is a quick hack to ensure the BadgerDB logs are captured by the Rossby logging system. This works well for right now but you can't tell what are DB logs and what are not, e.g. (on level `LogInfo`):

```
[rossby] 12:41:49.754287 All 0 tables opened in 0s
[rossby] 12:41:49.759921 Replaying file id: 0 at offset: 0
[rossby] 12:41:49.759986 Replay took: 49.017µs
[rossby] 12:41:49.760362 listening for requests on :1205
```

The first three messages are actually database messages and only the last message is a rossby message.

Reply:
Looks like the DB messages begin with capitalized words &mdash; subtle, but it's one way to identify where the messages are coming from without having to prepend anything specific to the DB messages to differentiate them.

Reply:
hehe - good point, I'll make sure I keep my log messages lower case until we can get something a bit more robust!